.github/workflows/on-pull-request.yml

```diff
@@ -44,3 +44,55 @@ jobs:
 
       - name: Test
         run: go test -v -race -p=1 -count=1
+  go-bench:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # to be able to retrieve the last commit in master branch
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version-file: './go.mod'
+          cache-dependency-path: './go.sum'
+          check-latest: true
+
+      - name: Run benchmark and store the output to a file
+        run: |
+          set -o pipefail
+          make bench | tee ${{ github.sha }}_bench_output.txt
+
+      - name: Get Main branch SHA
+        id: get-main-branch-sha
+        run: |
+          SHA=$(git rev-parse origin/main)
+          echo "sha=$SHA" >> $GITHUB_OUTPUT
+
+      - name: Get benchmark JSON from main branch
+        uses: actions/cache/restore@v3
+        with:
+          path: ./cache/benchmark-data.json
+          key: ${{ steps.get-main-branch-sha.outputs.sha }}-${{ runner.os }}-go-benchmark
+          fail-on-cache-miss: true
+
+      - name: Compare benchmarks
+        uses: benchmark-action/github-action-benchmark@v1
+        if: success()
+        with:
```

Comment:
Please add `alert-threshold` setting here so it's clear what the performance threshold is.